drivers/overlay/overlay.go

```diff
@@ -2087,11 +2087,22 @@ func (g *overlayFileGetter) Close() error {
 	return errs.ErrorOrNil()
 }
 
-func (d *Driver) getStagingDir(id string) string {
+// stagingDirForLayer returns the path to the base staging directory used for the given layer.
+func (d *Driver) stagingDirForLayer(id string) string {
```

Comment:
Ultimately `id` (“for the given layer”) here should not be relevant as an input (to `ApplyDiffFromStagingDirectory`), because it was not used as an input when creating the staging directory in `ApplyDiffWithDiffer`.

The computation of `stagingDirBase` in `stagingDirForLayer` is different — shouldn’t that be the same code, or maybe a single shared implementation?

Reply:
The difference is that here we know what is the id for the layer, so we validate that we are using a correct staging directory.  Not sure it is worth to keep it around, I am going to drop the check, if the staging directory is not valid (for any reason), the atomic rename will fail anyway

Reply:
My point really was that the `id` is not a factor on what should be a “correct” staging directory.

I don’t feel strongly about the the sanity check; removing it entirely does resolve the “different computation” concern.